docs/_scripts/gen_api_ref_pages.py

```diff
@@ -0,0 +1,32 @@
+"""Generate API reference pages."""
+
+from pathlib import Path
+
+import mkdocs_gen_files
+
+nav = mkdocs_gen_files.Nav()
+
+for path in sorted(Path("src").rglob("*.py")):
+    module_path = path.relative_to("src").with_suffix("")
+    doc_path = path.relative_to("src").with_suffix(".md")
+    full_doc_path = Path("reference", doc_path)
+
+    parts = list(module_path.parts)
+
+    if parts[-1] == "__init__":
+        parts = parts[:-1]
+        doc_path = doc_path.with_name("index.md")
+        full_doc_path = full_doc_path.with_name("index.md")
+    elif parts[-1] == "__main__":
+        continue
```

Comment:
I think this might be clearer as an `if` on `module_path.name`?

So let's say:

```python
if module_path.name == "__init__":
        # climb up one level for the module name
        module_path = module_path.parent
        doc_path = doc_path.with_name("index.md")
        full_doc_path = full_doc_path.with_name("index.md")
elif module_path.name == "__main__":
        continue
```

And then you continue below as normal with the now-correctly-truncated module path parts.

Reply:
See above. I've added a link to the original source for the script in 2300f44.

Reply:
Eh, I still feel like it's better the other way, but this discussion probably belongs to `mkdocs` then :p